packages/vitest/src/node/core.ts

```diff
@@ -84,12 +87,17 @@ export class Vitest {
 
     this.reporters = await createReporters(resolved.reporters, this.runner)
 
+    this.coverageReporter = options.coverage?.provider === 'istanbul' ? new IstanbulReporter() : new C8Reporter()
+    this.coverageReporter.initialize(this)
+
+    this.config.coverage = this.coverageReporter.resolveOptions()
+
     this.runningPromise = undefined
 
     this._onRestartListeners.forEach(fn => fn())
 
-    if (resolved.coverage.enabled)
-      await cleanCoverage(resolved.coverage, resolved.coverage.clean)
+    if (this.config.coverage.enabled)
+      await this.coverageReporter.clean(this.config.coverage.clean)
```

Comment:
Suggestion: create `NullCoverageProvider` (name is WIP :slightly_smiling_face: ) that does nothing and all `this.config.coverage.enabled` can be removed.

Reply:
This is now implemented in https://github.com/vitest-dev/vitest/pull/1676/commits/b699c0ce17e05b6a8c531901af635e977cd8f09f.